electron/src/helpers/BigGraph.tsx

```diff
@@ -0,0 +1,639 @@
+        });
+        uplotRef.current.setScale("y", {
+            min: minY - range * 0.1,
+            max: maxY + range * 0.1,
+        });
+    };
+
+    const setShowAllView = () => {
+        setViewMode('all');
+        manualScaleRef.current = null; // Clear manual scale
+
+        if (!uplotRef.current || !newData?.long) return;
+
+        // Get data from long buffer
+        const [timestamps, values] = seriesToUPlotData(newData.long);
+
+        if (timestamps.length === 0) return;
+
+        const fullStart = startTimeRef.current ?? timestamps[0];
+        const fullEnd = timestamps[timestamps.length - 1];
+        const minY = Math.min(...values);
+        const maxY = Math.max(...values);
+        const range = maxY - minY || 1;
+
+        uplotRef.current.setScale("x", {
+            min: fullStart,
+            max: fullEnd
+        });
+        uplotRef.current.setScale("y", {
+            min: minY - range * 0.1,
+            max: maxY + range * 0.1,
+        });
+    };
+
+    // Handle time window selection change
+    const handleTimeWindowChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+        const newTimeWindow = parseInt(event.target.value);
+        setSelectedTimeWindow(newTimeWindow);
+
+        // If currently in default view, immediately apply the new time window
+        if (viewMode === 'default') {
+            setDefaultView();
+        }
+    };
+
+    // Get the display value: cursor value if hovering, otherwise latest value
+    const displayValue = cursorValue !== null ? cursorValue : newData?.current?.value;
+
+    // Get the current time window label for the button
+    const currentTimeWindowLabel = TIME_WINDOW_OPTIONS.find(option => option.value === selectedTimeWindow)?.label || '1 min';
+
+    return (
+        <div style={{ width: "100%", height: "100%", display: "flex", flexDirection: "column" }}>
+            {/* Control Panel - Outside the graph */}
+            <div
+                style={{
+                    display: "flex",
+                    justifyContent: "space-between",
+                    alignItems: "center",
+                    padding: "8px",
+                    borderBottom: "1px solid #eee",
+                    backgroundColor: "#f9f9f9",
+                    flexShrink: 0,
+                }}
+            >
+                {/* Current Value Display - now shows cursor value when hovering */}
+                <div
+                    style={{
+                        background: "rgba(255, 255, 255, 0.9)",
+                        border: "1px solid #ccc",
+                        padding: "6px 12px",
+                        borderRadius: "4px",
+                        fontSize: "14px",
+                        fontWeight: "bold",
+                    }}
+                >
+                    Current: {displayValue !== undefined && displayValue !== null ? (renderValue ? renderValue(displayValue) : displayValue.toFixed(3)) : 'N/A'} {renderUnitSymbol(unit)}
```

Comment:
Please install a TypeScript formatter like prettier or something an reformat this file